models/maat/R/model2netcdf.MAAT.R

```diff
@@ -153,12 +165,27 @@ model2netcdf.MAAT <- function(rundir, outdir, sitelat = -999, sitelon = -999, st
                            newunits="Pa", missval=-999, longname="Leaf Internal CO2 Concentration", ncdims=ncdims)
       output <- var_update(sub.maat.output$cc, output, "cc", "Cc", oldunits="Pa", 
                            newunits="Pa", missval=-999, longname="Leaf Mesophyll CO2 Concentration", ncdims=ncdims)
+      
+      ## put in time_bounds before writing out new nc file
+      #length(output$var)
+      output$var[[length(output$var) + 1]] <- ncdf4::ncvar_def(name = "time_bounds", units = "", 
+                                                         longname = "history time interval endpoints", 
+                                                         dim = list(time_interval, time = t), 
+                                                         prec = "double")
+      output$dat[[length(output$dat) + 1]] <- c(rbind(bounds[, 1], bounds[, 2]))
```

Comment:
I'm not sure what having the output object formatted as `output$var$<somevariable>` and `output$dat$<somevariable>` gains you, but it causes you a lot of headache and awkward code that could easily by avoided by inverting the structure, such that you have `output$<somevariable>$dat/var`. Then, you could assign it with a single `list` call, and loop over it more simply later:

```r
output <- list(
  year = var_update(out.year, ...),
  leaf_assim = var_update(sub.maat.output$A, ...),
  leaf_resp = var_update(sub.maat.output$rd, ...),
  ...
)
# ...
for (out_var in output) {
  ncvar_put(ncout, out_var$avr, out_var$dat)
}
```